core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -0,0 +1,330 @@
+      return Integer.MAX_VALUE;
+    return (int) timeout;
+  }
+
+  @Override
+  public void setRange(Range range) {
+    this.range = range;
+  }
+
+  @Override
+  public Range getRange() {
+    return range;
+  }
+
+  @Override
+  public void setBatchSize(int size) {
+    this.batchSize = size;
+  }
+
+  @Override
+  public int getBatchSize() {
+    return batchSize;
+  }
+
+  @Override
+  public void enableIsolation() {}
+
+  @Override
+  public void disableIsolation() {}
+
+  @Override
+  public synchronized void setReadaheadThreshold(long batches) {
+    Preconditions.checkArgument(batches > 0);
+    readaheadThreshold = batches;
+  }
+
+  @Override
+  public synchronized long getReadaheadThreshold() {
+    return readaheadThreshold;
+  }
+
+  @Override
+  public Authorizations getAuthorizations() {
+    return opts.auths;
+  }
+
+  @Override
+  public void addScanIterator(IteratorSetting cfg) {
+    super.addScanIterator(cfg);
+  }
+
+  @Override
+  public void removeScanIterator(String iteratorName) {
+    super.removeScanIterator(iteratorName);
+  }
+
+  @Override
+  public void updateScanIteratorOption(String iteratorName, String key, String value) {
+    super.updateScanIteratorOption(iteratorName, key, value);
+  }
+
+  private static final Range EMPTY_RANGE = new Range();
+
+  private class IterEnv extends BaseIteratorEnvironment {
+    @Override
+    public IteratorScope getIteratorScope() {
+      return IteratorScope.scan;
+    }
+
+    @Override
+    public boolean isFullMajorCompaction() {
+      return false;
+    }
+
+    @Override
+    public Authorizations getAuthorizations() {
+      return opts.auths;
+    }
+
+    @Override
+    public boolean isSamplingEnabled() {
+      return RFileScanner.this.getSamplerConfiguration() != null;
+    }
+
+    @Override
+    public SamplerConfiguration getSamplerConfiguration() {
+      return RFileScanner.this.getSamplerConfiguration();
+    }
+  }
+
+  @Override
+  public Iterator<Entry<Key,Value>> iterator() {
+    try {
+      RFileSource[] sources = opts.in.getSources();
+      List<SortedKeyValueIterator<Key,Value>> readers = new ArrayList<>(sources.length);
+      for (int i = 0; i < sources.length; i++) {
+        FSDataInputStream inputStream = (FSDataInputStream) sources[i].getInputStream();
+        readers.add(new RFile.Reader(new CachableBlockFile.Reader(inputStream, sources[i].getLength(), opts.in.getConf(), dataCache, indexCache,
+            AccumuloConfiguration.getDefaultConfiguration())));
+      }
+
+      if (getSamplerConfiguration() != null) {
+        for (int i = 0; i < readers.size(); i++) {
+          readers.set(i, ((Reader) readers.get(i)).getSample(new SamplerConfigurationImpl(getSamplerConfiguration())));
+        }
+      }
+
+      SortedKeyValueIterator<Key,Value> iterator;
+      if (opts.lowerBound != null || opts.upperBound != null) {
+        iterator = new MultiIterator(readers, new Range(opts.lowerBound, false, opts.upperBound, true));
```

Comment:
> At a glance, it looked like the lowerBound and upperBound attributes on the opts variable were losing "precision" over what is usually possible via an end-user's use of Range directly. 

Ok.   When I wrote the API methods I thought that MultiIterator only supported row ranges.  Then I got there and found otherwise, but left it as is.  Based on your comment I am thinking maybe there should just be a `setBounds(Range range)` method.